core/controllers/classifier.py

```diff
@@ -123,3 +126,31 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+class NextJobHandler(base.BaseHandler):
+    """ This handler fetches next job in job queue and sends back job_id,
+    algorithm_id and training data to the VM.
+    """
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    def post(self):
+        """ Handels POST requests. """
+        signature = self.payload.get('signature')
+        vm_id = self.payload.get('vm_id')
+        if vm_id == feconf.DEFAULT_VM_ID and not feconf.DEV_MODE:
+            raise self.UnauthorizedUserException
+        if not verify_signature(None, vm_id, signature):
+            raise self.UnauthorizedUserException
+        response = {}
```

Comment:
Add a newline above this one. In general, use newlines judiciously to group your code into chunks that are conceptually-related. Don't just write one big blob of code -- that's quite hard to read.